Improve the json schema error response

jayrbolton opened the pull request:
Instead of the returning the whole schema, I would rather see just the path to the data that had the error along with the validation error message.

MrCreosote commented:
The changes are minor and I don't see any issues re syntax etc, but I don't really understand the changes. What's an `err`?

MrCreosote commented:
I guess more specifically, what do the fields of `err` mean is the question I'm asking

MrCreosote commented:
Could you explain the `ValidationError` fields?

jayrbolton commented:
The jsonschema.exception.ValidationError is explained pretty well here: https://python-jsonschema.readthedocs.io/en/stable/errors/

I will add a link to these docs in the comments

MrCreosote commented on src/relation_engine_server/main.py:
```diff
@@ -76,11 +76,11 @@ def view_does_not_exist(err):
         'validator_value': err.validator_value,
-        'schema': err.schema
+        'path': list(err.absolute_path),
+        'schema_path': list(err.schema_path)
```
Should this be `absolute_schema_path`?

MrCreosote commented on src/relation_engine_server/main.py:
```diff
@@ -76,11 +76,11 @@ def view_does_not_exist(err):
 def validation_error(err):
     """Json Schema validation error."""
     resp = {
```
Is the error structure returned from the re api documented somewhere? Looking at the test, it was hard for me to figure out what the fields mean.

jayrbolton commented:
I added some docs in the readme as well as a comment about the error class in the code.

Also I removed the "schema path" field in the response as I think that'll be more confusing than useful.

MrCreosote commented:
Note that based on the diff
* `schema_path` is still in the structure
* the documentation references a `value` field when the actual field is `validator_value`

jayrbolton commented:
Good catch, forgot to push it all
